src/intrinsics/ecma262/ArrayPrototype.js

```diff
@@ -697,9 +728,23 @@ export default function(realm: Realm, obj: ObjectValue): void {
 
   // ECMA262 22.1.3.16
   obj.defineNativeMethod("map", 1, (context, [callbackfn, thisArg]) => {
-    // 1. Let O be ? ToObject(this value).
+    // 1. Let O be ? ToObject(this value).]
     let O = To.ToObject(realm, context);
 
+    if (Widen.hasWidenedNumericUnknownProperty(O)) {
```

Comment:
I think we might need to check if this is a simple object which it should be.

This is safe to do on anything that is a isSimpleObject with unknown properties. Doesn't have to be numeric specifically. One approach could be to just try executing the full path and if it throws a FatalError and is a simple object, then do this.